.azure-pipelines/templates/build.yml

```diff
@@ -0,0 +1,87 @@
+parameters:
+- name: py_version
+  type: string
+  default: ''
+- name: image
+  type: string
+  default: ''
+- name: ccache
+  type: boolean
+  default: false
+- name: OSX_VERSION
+  type: string
+  default: ''
+- name: publish
+  type: string
+  default: ''
```

Comment:
I think in the case of `OSX_VERSION` it has to be left blank, because we test further down the pipeline whether the variable is set or not.
So it seemed consistent/sensible to do the same for the others.
But I am happy to change if you have good suggestions for invalid defaults.